xgcm/test/test_transform.py

```diff
@@ -892,6 +892,15 @@ def test_grid_transform_multidim(request, client, multidim_cases):
 
     na = 3
     source = source * xr.DataArray(np.ones([na]), dims=["a"])
+    # broadcast the target_data manually
+    print(transform_kwargs)
+    target_data = transform_kwargs.pop("target_data", None)
+    print(target_data)
+    if not target_data is None:
+        target_data = target_data * xr.DataArray(np.ones([na]), dims=["a"])
+        if client != "no_client":
+            target_data = target_data.chunk({"a": 1})
```

Comment:
Something about this line must be breaking things but I don't see how.

Reply:
Yeah it seems really freaking odd to me. Is there a sleeker way to broadcast the same array along a new axis n times in xarray?

Reply:
`target_data = target_data.expand_dims(a=3)`

Reply:
oh you have a view from broadcasting, that's the problem; `target_data = target_data.expand_dims(a=3).copy(deep=True)` may work

Reply:
Awesome, thank you so much @dcherian. 

The problem was definitely related to an inplace modification of the data, not dask (I switched the scheduler order and it happend for the first test of each fixture).
The solution was to change the scope of the fixtures (I had it set to 'module'). Then it works.